### PR TITLE
Update Batch File Conventions version for republish as strong named DLL

### PR DESCRIPTION
**Pull request by itowlson**
## Description

Updates Batch.FileConventions assembly and NuGet package version for the publication of a signed (strong named) version of the DLL.
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/dev/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as a link to the swagger spec, used to generate the code.
- [ ] The `project.json` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@itowlson__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

**Comment by matthchr** [comment]
@itowlson This looks good to me.

**Review by matthchr — APPROVED** [review]

**Comment by itowlson** [comment]
@azurecla Um, I don't think a CLA is required. I'm a Microsoft employee and part of the Azure organisation...

File: src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -19,8 +19,8 @@
 [assembly: AssemblyTitle("Azure Batch File Conventions")]
 [assembly: AssemblyDescription("A convention-based library for saving and retrieving Azure Batch task output files.")]
 
-[assembly: AssemblyVersion("1.0.0.0")] 
-[assembly: AssemblyFileVersion("1.0.0.0")]
+[assembly: AssemblyVersion("1.0.1.0")]
```
Comment (markcowl) [inline src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs]: You should not change the assembly version, for backward compatibility

**Review by markcowl — CHANGES_REQUESTED** [review]

File: src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (matthchr) [inline src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs]: Can you explain this a bit more @markcowl? Why should we not change the assembly version?

Comment (matthchr) [inline src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs]: Pinging on this @markcowl -- what's the reasoning here?

Comment (markcowl) [inline src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs]: This is our general guidance for versioning.  Using AssemblyVersion = 'Major.0.0.0', we allow additive updates of the assembly without recompiling or requiring an assembly redirect for programs that use this assembly.  It is useful for scenarios like PowerShell or other library usages, where a config file with assembly redirects is not practical.

Comment (markcowl) [inline src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs]: However, it is proper for the AssemblyFileVersion to match the package version - this enables upgrade installation scenarios

Comment (itowlson) [inline src/Batch/FileConventions/Src/AzureBatchFileConventions/Properties/AssemblyInfo.cs]: @markcowl Reverted AssemblyVersion as requested.

**Comment by itowlson** [comment]
@markcowl Could you re-review please and let us know if any further changes are needed - we have a customer asking pressing us for an updated package.

**Review by markcowl — APPROVED** [review]

**Pull request merged**